batsat/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! # BatSat Solver Interface
+//!
+//! Interface to the [BatSat](https://github.com/c-cube/batsat) incremental SAT-Solver
+
+use batsat::{intmap::AsIndex, lbool, BasicSolver, SolverInterface};
+use rustsat::{
+    solvers::{Solve, SolveIncremental, SolverResult},
+    types::{Clause, Lit, TernaryVal},
+};
+use thiserror::Error;
+
+#[derive(Error, Clone, Copy, PartialEq, Eq, Debug)]
+#[error("BatSat returned an invalid value: {error}")]
+pub struct InvalidApiReturn {
+    error: &'static str,
+}
+
+pub struct BatsatBasicSolver(BasicSolver);
```

Comment:
To add a little bit more to my comment from earlier, implementing this interface either in RustSAT or in BatSat directly would eliminate the need for this wrapper type. This type is basically only needed because one can only implement local trait or foreign traits for local types.
I am still unsure whether I think that implementing interfaces like this in RustSAT or in the solver crate itself should be preferred. That of course also strongly depends on whether the solver developers would be interested in supporting RustSAT directly.

Reply:
I agree, there are these 2 ways to approch this. To be hones, for me it is not completely understandable why rust is not allowing to extend an existing struct from a other crate with new traits, but the creators will have had their reasons. 
As far as I have researched, the prefered way to deal with adding traits to a existing structs is using these wrappers. In terms of computational overhead I pretty sure this is neglectable, since the compiler will elimiate the wrappers in during the code optimization.

I will reach out to the batsat developers to ask how they think about it.